Sources/Workspace/Destination.swift

```diff
@@ -56,7 +56,9 @@ public struct Destination {
     /// - Parameter originalWorkingDirectory: The working directory when the program was launched.
     private static func hostBinDir(
         originalWorkingDirectory: AbsolutePath? = localFileSystem.currentWorkingDirectory
-    ) -> AbsolutePath {
+    ) throws -> AbsolutePath {
+      #if COMPILE_FOR_TOOLCHAIN // Compiling as part of the Swift project.
```

Comment:
Instead of using #if for this, we could dynamically check determine if we're installed in the toolchain. If not, then proceed to lookup the tools using xcrun and which.

Reply:
Alright. I’ll do that then.

Reply:
The check is now dynamic.